enhancements/windows-containers/container-runtime-containerd.md

```diff
@@ -0,0 +1,205 @@
+do not allow switching between runtime.
+
+## Containerd logging
+Containerd can be started with parameters in which we can enable logging and specify the file path to
+log warnings/errors. Log files will be stored at c:\var\log\containerd. Pod logs will be stored in the
+same location (C:\var\log\pods)
+
+## Upgrade
+WMCO supports upgrade. Please refer [WMCO-Upgrade](https://github.com/openshift/enhancements/blob/master/enhancements/windows-containers/windows-machine-config-operator-upgrades.md)
+enhancement proposal document to learn more about WMCO upgrade.
+
+The procedure for an upgrade is as follows:
+1) MachineSet
+   * As part of upgrade Machine object gets deleted, resulting in the drain and deletion of the Windows node.
+   * New Windows VM will be created.
+   * The upgraded WMCO instance will configure Windows VM with containerd as the default runtime.
+   * Once the Windows node joins the Kubernetes cluster, the user can deploy pods on the Windows node.
+2) BYOH ( Bring Your Own Host)
+   * upgrade process will uninstall kubelet, kube-proxy, CNI, and hybrid-overlay components that were installed by WMCO.
+   * Windows OS-specific configurations like HNS-Network that were created by WMCO instance would be deleted.
+   * Once cleanup is done, WMCO instance will install containerd along with kubelet, kube-proxy, CNI and
+     hybrid-overlay.
+   * Kuebelet will start as a service. Containerd will be the default runtime.
+   * Once the Windows node joins the Kubernetes cluster, the user can deploy pods on the Windows node.
+
+### Risks and Mitigations
+
+* If the cluster is upgraded, and the new version introduces an issue due to containerd
+  that should be addressed because Docker runtime support won't be available in kubelet.
+  As we don't support downgrade, reverting to an older version is not possible. To overcome
+  any issues, we introduce this feature as part of the 5.0 community operator so that this feature
+  can be well tested before we make it default in WMCO 6.0 release. We will use two test jobs to test all e2e test cases
+  in the 4.10 community branch(OKD) and 4.10 branch(ccm OCP)to capture any regression.
+  This will help us to identify any issues well before containerd becomes default runtime.
+* If we run into any issue in containerd, we work closely with the containerd open source community and try
+  to resolve it as soon as possible.
+* containerd doesn't support image-pull-progress-deadline as of now. There is a PR
+  https://github.com/containerd/containerd/pull/6150 work in progress. Until this
+  gets merged, if Windows image pull takes more time than the default value, we might
+  run into to image pull timeout error. To overcome this issue, the user should log in to Windows VM and install `ctr or crictl`
+  command-line tool. Users can use one of these tools to pull the container image manually. Upon successful image download,
```

Comment:
We shouldnt be suggesting this workflow. The user would have to do this everytime they spin up a VM. Instead we should be recommending they bake the required container images into the VM image they are using.

Reply:
That's not the right solution. How would customers know in advance what are all containers images they will be using in the future. There may be different applications they would want to run in a different container.  They can build the new container image after they bring up Window VM. I hope by the time we enable contained as the default runtime, this issue would have been fixed in the upstream.

Reply:
@selansen there is no harm in suggesting to the user that they bake in the base images. These base images will be needed irrespective of their particular application images given they need a base layer to build from.

Reply:
I can agree with baking some base images but, how will you expect customers to know all the images that will be used in the windows node in advance? What if customers want to run a new container image once VM is up and running?
I shall add it as an alternate solution but still feel this is not the right workaround.

Reply:
>I shall add it as an alternate solution but still feel this is not the right workaround.

Baking in base images is not an alternate solution. This IMO is the correct workaround. 

>What if customers want to run a new container image once VM is up and running?

This new container image will also need to be built off one of the existing base images so having it baked in will help.